File: Other Algorithms/Kadane Algorithm/java/Kadane.java
class Kadane { 
  
    static int maxSubArraySum(int a[], int size) { 
        int max_so_far = a[0]; 
        int curr_max = a[0]; 
      
        for (int i = 1; i < size; i++) 
        { 
               curr_max = Math.max(a[i], curr_max+a[i]); 
            max_so_far = Math.max(max_so_far, curr_max); 
        } 
        return max_so_far; 
    } 
  

    public static void main(String[] args) { 
        int a[] = {-2, -3, 4, -1, -2, 1, 5, -3}; 
        int n = a.length;    
        int max_sum = maxSubArraySum(a, n); 
        System.out.println("Maximum contiguous sum is " + max_sum); 
    } 
}

File: kinhosz codeforces/kadane.cpp
vector<int> lista;

int kadane(){

  int max_aqui = 0;
  int max_total = 0;
  
  for(int i=0;i<lista.size();i++){
  
     max_aqui += lista[i];
     
     if(max_aqui < 0) max_aqui = 0;
     max_total = max(max_total,max_aqui);
  }
  return max_total;
}
